src/Psalm/Internal/LanguageServer/Server/TextDocument.php

```diff
@@ -92,8 +90,9 @@ public function didOpen(TextDocumentItem $textDocument): void
      * The document save notification is sent from the client to the server when the document was saved in the client
      *
      * @param TextDocumentItem $textDocument the document that was opened
+     * @param ?string $text the content when saved
      */
-    public function didSave(TextDocumentItem $textDocument): void
+    public function didSave(TextDocumentItem $textDocument, ?string $text): void
```

Comment:
null won't happen in this case, because `includeText = true`
https://github.com/vimeo/psalm/blob/e91e33c39c2d15f326f91da7906744714b387a65/src/Psalm/Internal/LanguageServer/LanguageServer.php#L265
Should I keep it for consistency with the language server protocol specification?
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#didSaveTextDocumentParams

Reply:
Yea. It should match the LSP spec. Just because include text is true does not mean a manage server will respect that

Reply:
This will help psalm to analyze the file on save, even if we use `disable-on-change`